mlflow/store/artifact/databricks_artifact_repo.py

```diff
@@ -722,9 +771,24 @@ def _download_file(self, remote_file_path, local_path):
         # Read credentials for only one file were requested. So we expected only one value in
         # the response.
         assert len(read_credentials) == 1
-        self._download_from_cloud(
-            cloud_credential_info=read_credentials[0], dst_local_file_path=local_path
-        )
+
+        # list_artifacts API only returns a list of FileInfos at the specified path
+        # if it's a directory. To get file size, we need to iterate over FileInfos
+        # contained by the parent directory. A bad path could result in there being
+        # no matching FileInfos (by path), so fall back to None size to prevent
+        # parallelized download.
+        parent_dir, _ = posixpath.split(remote_file_path)
```

Comment:
```suggestion
        parent_dir = posixpath.dirname(remote_file_path)
```

Can we use `dirname` here?